### DISCUSSION TITLE
Detect and show if a smart contract is upgradable or not (e. g. in Casper Wallet)

### DISCUSSION THREAD

**Discussion by GuybrushX** [casper-network/casper-node#4047] [Updates]
Casper supports upgradable and immutable smart contracts depending on the use-case.

After a short discussion in Discord with a SRE it seems that right now there is no flag available which can be queried programmatically to see if a smart contract is upgradable or not.

I also saw this question from a community member months ago already to show this e. g. in the Casper Wallet and I think it makes sense to add a little bit more security and control to the user.

In other words:
If a project claims that the used smart contracts are immutable it's not possible to confirm this. That means that currently it is possible to silently upgrade the smart contract which changes essential behavior or does other malicious things which a user would not agree to.

I'm aware that you can interact with a specific version of a contract only but this doesn't help the user where the contract calls are made by the dApp and not the user.

I would like to get this officially confirmed and if this is something which could or should be added?

**Comment by KillianH** [comment]
https://github.com/casper-network/casper-node/releases/tag/v1.5.1

"Add a lock_status field to the JSON representation of the ContractPackage values." Should do the trick :)

**Reply by GuybrushX** [reply]
Good catch - it could be that this is the right flag but to be honest I have no idea what it really represents.

Maybe @rafal-ch can shed some light here?
